django/db/models/fields/related_descriptors.py

```diff
@@ -85,13 +85,11 @@ def __set__(self, instance, value):
 
 def _filter_prefetch_queryset(queryset, field_name, instances):
     predicate = Q(**{f"{field_name}__in": instances})
-    if queryset.query.is_sliced:
+    db = queryset._db or DEFAULT_DB_ALIAS
+    if queryset.query.is_sliced and connections[db].features.supports_over_clause:
```

Comment:
Any thoughts about error'ing out with a clearer error message than `AssertionError: Cannot filter a query once a slice has been taken.` when `not supports_over_clause` instead? Maybe `NotSupportedError("Prefetching from a limited queryset is only supported on backends that support window functions")`?

Reply:
Added